string t_name
string r_name
float64 t_ex